Softeq.XToolkit.WhiteLabel/Navigation/FluentNavigators/FrameFluentNavigator.cs

```diff
@@ -8,42 +8,103 @@
         {
-            _frameNavigationService = navigationService;
+            _navigationService = navigationService
+                ?? throw new ArgumentNullException(nameof(navigationService));
         }
 
-        public FrameFluentNavigator<TViewModel> WithParam<TValue>(Expression<Func<TViewModel, TValue>> property,
+        /// <summary>
+        ///     Creates and adds navigation parameter for the specified property using reflection.
+        /// </summary>
+        /// <typeparam name="TValue">Type of parameter value.</typeparam>
+        /// <param name="propertyExpression">Target property.</param>
+        /// <param name="value">Value to set.</param>
+        /// <returns>Self.</returns>
+        /// <exception cref="T:System.ArgumentNullException">
+        ///     <paramref name="propertyExpression"/> cannot be <see langword="null"/>.
+        /// </exception>
+        public FrameFluentNavigator<TViewModel> WithParam<TValue>(
+            Expression<Func<TViewModel, TValue>> propertyExpression,
             TValue value)
         {
-            ApplyParameter(property, value);
+            if (propertyExpression == null)
+            {
+                throw new ArgumentNullException(nameof(propertyExpression));
+            }
+
+            ApplyParameter(propertyExpression, value);
             return this;
         }
 
-        public FrameFluentNavigator<TViewModel> WithParams(IEnumerable<NavigationParameterModel> navigationHelperParameters)
-        {
-            Parameters.AddRange(navigationHelperParameters);
+        /// <summary>
+        ///     Adds multiple navigation parameters to the current instance.
+        /// </summary>
+        /// <param name="navigationParameters">List of navigation parameters to add.</param>
+        /// <returns>Self.</returns>
+        /// <exception cref="T:System.ArgumentNullException">
+        ///     <paramref name="navigationParameters"/> cannot be <see langword="null"/>.
+        /// </exception>
+        public FrameFluentNavigator<TViewModel> WithParams(IEnumerable<NavigationParameterModel> navigationParameters)
+        {
+            if (navigationParameters == null)
+            {
+                throw new ArgumentNullException(nameof(navigationParameters));
+            }
+
+            Parameters.AddRange(navigationParameters);
             return this;
         }
 
+        /// <summary>
+        ///     Initializes current service with the specified <paramref name="initParameter"/>.
+        /// </summary>
+        /// <param name="initParameter">
+        ///      Navigation object (usually platform-specific) that is used for initialization.
+        /// </param>
+        /// <returns>Self.</returns>
         public FrameFluentNavigator<TViewModel> Initialize(object initParameter)
         {
-            _frameNavigationService.Initialize(initParameter);
+            _navigationService.Initialize(initParameter);

```

Comment:
Seems, need null-heck for `initParameter`

Reply:
Fixed